graph_dfs_debug/graph.py

```diff
@@ -19,36 +20,40 @@ def add_vertex(self, vertex, edges=()):
         self.vertices[vertex] = set(edges)
 
     def add_edge(self, start, end, bidirectional=True):
-        self.vertices[start].add(start)
+        self.vertices[start].add(end)
         if bidirectional:
-            self.vertices[end].add(end)
+            self.vertices[end].add(start)
 
```

Comment:
What happens is neither start nor end are in vertices? You need to handle that somewhow